src/main/scala/firrtl/Emitter.scala

```diff
@@ -979,6 +979,10 @@ class VerilogEmitter extends SeqTransform with Emitter {
         }
         emit(Seq("`endif"))
         emit(Seq("`ifndef SYNTHESIS"))
+        // User-defined macro of code to run before an initial block
+        emit(Seq("`ifdef BEFORE_INITIAL"))
+        emit(Seq("`BEFORE_INITIAL"))
+        emit(Seq("`endif"))
```

Comment:
We didn't do this for the `RANDOMIZE` stuff, but should we prefix our macros with `FIR` or `FIRRTL` just in case FIRRTL-generated Verilog is being used along side other Verilog that may have it's own macros?

Reply:
I also think this is a good idea, so I updated the macros specific to this PR and opened up issue #1554 to prompt discussion for the other macros.

Reply:
oh i left my comment on #1554 before I read this discussion, as you were :)